src/vm/comsynchronizable.cpp

```diff
@@ -1537,15 +1537,56 @@ void QCALLTYPE ThreadNative::nativeInitCultureAccessors()
     END_QCALL;
 }
 
+typedef HRESULT(WINAPI *pfnSetThreadDescription)(HANDLE hThread, PCWSTR lpThreadDescription);
+extern pfnSetThreadDescription g_pfnSetThreadDescription;
+BOOL g_InitializeSetThreadDescriptionWasRun = FALSE;
+
+
+HRESULT WINAPI InitializeSetThreadDescription(HANDLE hThread, PCWSTR lpThreadDescription)
+{
+    pfnSetThreadDescription func = NULL;
+
+#ifndef FEATURE_PAL
+    if (!g_InitializeSetThreadDescriptionWasRun)
+    {
+        HMODULE hKernel32 = WszLoadLibrary(W("kernel32.dll"));
+        if (hKernel32 != NULL)
+        {
+            func = (pfnSetThreadDescription)GetProcAddress(hKernel32, "SetThreadDescription");
+        }
+        g_InitializeSetThreadDescriptionWasRun = TRUE;
```

Comment:
@Alois-xx I think @jkotas means that when the function is available, the second time InitializeSetThreadDescription is called, it will do nothing (rather than return the function). Although I would expect InitializeSetThreadDescription to be called only once, my C++ is very rusty.